hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/InternalValueReader.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.serialization.impl;
+
+import com.hazelcast.query.extractor.ValueReader;
+
+import java.io.IOException;
+
+public interface InternalValueReader extends ValueReader {
+
+    Object read(String fieldPath) throws IOException;
```

Comment:
How this method is different than `ValueReader.read(path, collector)`? What is the use-case?